### PR TITLE
Support block queries using block height as block identifier

### LINKED ISSUE #94: Support querying using block height
Support using block height as identifier for blocks in queries.

Related to https://github.com/Concordium/concordium-node/issues/837.

### PR DESCRIPTION
**Pull request by limemloh**
## Purpose

Closes #94
Depends on https://github.com/Concordium/concordium-base/pull/374.

## Changes

- Update base and implement gRPC changes.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: examples/v2_get_block_info.rs [hunk 1]
```diff
@@ -39,5 +40,21 @@ async fn main() -> anyhow::Result<()> {
...
+    {
+        let identifier = v2::RelativeBlockHeight {
+            genesis_index: 0.into(),
+            height:        0.into(),
+            restrict:      true,
+        };
+        let ai = client.get_block_info(identifier).await?;
+        println!("Block at relative height {:?} {:#?}", identifier, ai);
+    }
+
```
Comment (Bargsteen) [inline examples/v2_get_block_info.rs:58]: Consider adding optional parameters to `App` for the absolute and relative height, such that it is easy to test whether it works for other values than 0. Especially when FFI is involved in the node ;)

File: src/v2/conversions.rs [hunk 2]
```diff
@@ -2387,6 +2410,9 @@ impl TryFrom<BlockInfo> for super::types::queries::BlockInfo {
             era_block_height:        value.era_block_height.require()?.into(),
             genesis_index:           value.genesis_index.require()?.into(),
             block_baker:             value.baker.map(|b| b.into()),
+            protocol_version:        ProtocolVersion::from_i32(value.protocol_version)
+                .ok_or_else(|| tonic::Status::internal("Unknown protocol version"))?
```
Comment (Bargsteen) [inline src/v2/conversions.rs:2414]:
```suggestion
                .ok_or_else(|| tonic::Status::internal("Unknown protocol version: {}"), value.protocol_version)?
```

File: src/v2/mod.rs [hunk 3]
```diff
@@ -116,6 +116,25 @@ pub enum BlockIdentifier {
     LastFinal,
     /// Query in the context of a specific block hash.
     Given(BlockHash),
+    /// Query for a block at absolute height, if a unique block can be
+    /// identified at that height.
+    AbsoluteHeight(AbsoluteBlockHeight),
+    /// Query for a block at height relative to genesis index, if a unique block
+    /// can be identified at that height.
```
Comment (Bargsteen) [inline src/v2/mod.rs:123]:
```suggestion
    /// Query for a block at a height relative to the genesis index. Only works if a unique block
    /// can be identified at that height.
```

File: src/v2/mod.rs [hunk 4]
```diff
@@ -116,6 +116,25 @@ pub enum BlockIdentifier {
     LastFinal,
     /// Query in the context of a specific block hash.
     Given(BlockHash),
+    /// Query for a block at absolute height, if a unique block can be
```
Comment (Bargsteen) [inline src/v2/mod.rs:119]:
```suggestion
    /// Query for a block at absolute height, i.e. the height from the original genesis. Only works if a unique block can be
```

**Review by Bargsteen — APPROVED** [review]
Looks good. I just had some minor suggestions.

File: examples/v2_get_block_info.rs (hunk 1 above)
Comment (abizjak) [reply examples/v2_get_block_info.rs:58]: Yes please.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,7 @@
 ## Unreleased changes
 
+- Support relative and absolute block height as the block identifier in block queries.
+- Add field `protocol_version` to `BlockInfo` which is the protocol version of the queried block.
```
[the remark this answers is not in the document]
Comment (abizjak) [reply CHANGELOG.md:4]: And also change this in the README where is says "supported node version"

File: src/v2/mod.rs (hunk 3 above)
Comment (abizjak) [reply src/v2/mod.rs:123]: Revised.

File: src/v2/mod.rs (hunk 4 above)
Comment (abizjak) [reply src/v2/mod.rs:119]: Revised

File: CHANGELOG.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (abizjak) [reply CHANGELOG.md:4]: Changed.

File: src/v2/conversions.rs (hunk 2 above)
Comment (abizjak) [reply src/v2/conversions.rs]: Changed.

**Pull request merged**